File: docs/legacy(old PACE)/C#/4.0/PaceAPISample/sample/InvokePaceConnect.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using efipaceservices;
using System.IO;

namespace Pace_Web_Service_SDK.sample
{
    class InvokePaceConnect
    {
        public static void Run()
        {
            Console.WriteLine("Enter the pace connect id");
            String connectId = Console.ReadLine();

            Console.WriteLine("Enter the file name");
            String connectInput = readInputFromFile(Console.ReadLine());
            InvokePaceConnectHttpBinding invokePaceConnectHttpBinding = PaceClient.getInvokePaceConnectHttpBinding();

            try
            {
                ProcessResults results = invokePaceConnectHttpBinding.invokePaceConnect(connectId, connectInput);

                SuccessProcessItem[] successes = results.successes;
                if (null != successes)
                {
                    String str = "";
                    foreach (SuccessProcessItem successProcessItem in successes)
                    {
                        str += "\n\t" + successProcessItem.reason;
                    }

                    Console.WriteLine("Successes: " + str);
                }

                FailedProcessItem[] failures = results.failures;

                if (null != failures)
                {
                    String str = "";
                    foreach (FailedProcessItem failureProcessItem in failures)
                    {
                        str += "\n\t" + failureProcessItem.reason;
                    }

                    Console.WriteLine("Failures: " + str);
                }

            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
        }
        public static String readInputFromFile(String fileName)
        {
            String fileInput = "";
            if (!File.Exists(fileName))
            {
                Console.WriteLine("{0} does not exist!", fileName);
                throw new FileNotFoundException("File {0} entered not found",fileName);
            }
            else
            {
                FileStream fileStream = new FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.Read);
                using (StreamReader streamReader = new StreamReader(fileStream))
                {
                    string input = "";
                    while (streamReader.Peek() > -1)
                    {
                        input = streamReader.ReadLine();
                    }
                    fileInput = input;
                }
                return fileInput;
            }
        }
    }
}

File: docs/legacy(old PACE)/unsupported/VB.NET/PaceWebServiceSDK/sample/InvokePaceConnect.vb
Imports PaceWebServiceSDK.efipaceservices
Imports System.IO

Public Class InvokePaceConnect
    Public Shared Sub Run()
        Console.WriteLine("Enter the pace connect id")
        Dim connectId As String = Console.ReadLine()

        Console.WriteLine("Enter the file name")
        Dim connectInput As String = readInputFromFile(Console.ReadLine())
        Dim invokePaceConnectHttpBinding As InvokePaceConnectHttpBinding = PaceClient.getInvokePaceConnectHttpBinding()

        Try
            Dim results As ProcessResults = invokePaceConnectHttpBinding.invokePaceConnect(connectId, connectInput)

            Dim successes As SuccessProcessItem() = results.successes

            If successes IsNot Nothing Then

                Dim str As [String] = ""
                For Each successProcessItem As SuccessProcessItem In successes
                    str += vbLf & vbTab + successProcessItem.reason
                Next
                Console.WriteLine("Successes: " & str)
            End If

            Dim failures As FailedProcessItem() = results.failures

            If failures IsNot Nothing Then

                Dim str As [String] = ""
                For Each failureProcessItem As FailedProcessItem In failures
                    str += vbLf & vbTab + failureProcessItem.reason
                Next
                Console.WriteLine("Failures: " & str)
            End If

        Catch ex As Exception

        End Try

    End Sub

    Public Shared Function readInputFromFile(ByVal fileName As [String]) As [String]
        Dim fileInput As [String] = ""
        If Not File.Exists(fileName) Then
            Console.WriteLine("{0} does not exist!", fileName)
            Throw New FileNotFoundException("File {0} entered not found", fileName)
        Else
            Dim fileStream As New FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.Read)
            Using streamReader As New StreamReader(fileStream)
                Dim input As String = ""
                While streamReader.Peek() > -1
                    input += streamReader.ReadLine()
                End While
                fileInput = input
            End Using
            Return fileInput
        End If
    End Function


End Class
